jvm/core/src/main/scala/org/scalatest/tools/Framework.scala

```diff
@@ -886,8 +885,23 @@ import java.net.{ServerSocket, InetAddress}
               case e: RunCompleted => // Sub-process completed, just let the thread terminate
               case e: RunStopped => dispatchReporter(e)
               case e: RunAborted => dispatchReporter(e)
-	        }
+            }  
           }
+
+          @annotation.tailrec
+          final def tryReact(): Unit = 
+            try {
+              react()  
+            }
+            catch {
+              case t: Throwable => 
+                // Restart server socket
+                println(Resources.unableToReadSerializedEvent)
+                is.get.close()
+                socket.set(server.accept())
+                is.set(new SkeletonObjectInputStream(socket.get.getInputStream, getClass.getClassLoader))
+                tryReact()
+            }
```

Comment:
@cheeseng Should this loop forever in case it keeps getting an exception? At some point maybe it should exit?